### ISSUE TITLE
Inline floating point formatting shouldn't depend on default Locale

### ISSUE THREAD

**Issue by glandais** [jOOQ/jOOQ#12345]
### Expected behavior

Scientific notation used to format float/double values should not be dependent of the JVM locale.

### Steps to reproduce the problem

In https://github.com/glandais/jOOQ-mcve :

- locale set to Locale.FRANCE
- using SQLite, removing generated code & liquibase
- statement type set to STATIC_STATEMENT
- creation of a table and trying to insert rows

Error :

```
org.jooq.exception.DataAccessException: SQL [insert into test (id, value1, value2) values (100, 0E0, 0E0), (101, 1,45E1, 3,141592653589793E0), (102, 2,9E1, 6,283185307179586E0), (103, 4,35E1, 9,42477796076938E0), (104, 5,8E1, 1,2566370614359172E1), (105, 7,25E1, 1,5707963267948966E1), (106, 8,7E1, 1,884955592153876E1), (107, 1,015E2, 2,1991148575128552E1), (108, 1,16E2, 2,5132741228718345E1), (109, 1,305E2, 2,8274333882308138E1)]; [SQLITE_ERROR] SQL error or missing database (all VALUES must have the same number of terms)
```

### Versions

- jOOQ: 3.15.1
- Java:
- Database (include vendor):
- OS:
- JDBC Driver (include name if inofficial driver):

**Comment by glandais-sparklane** [comment]
Correct repo is https://github.com/glandais-sparklane/jOOQ-mcve

**Comment by glandais-sparklane** [comment]
Linked to #11711

**Comment by lukaseder** [comment]
Thanks a lot for your report. Indeed, this was introduced with #11711 in jOOQ 3.15. This default locale keeps producing unfortunate behaviour... The fix is to be explicit about using the `Locale.US` here:

```java
new DecimalFormat("0.################E0", DecimalFormatSymbols.getInstance(Locale.US))
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12352).

Thanks again for your report. Nice issue number, btw!
